File: phonebook.pl
#!/usr/bin/perl -w
# Jadach Dominika


use strict;
use warnings;
use Scalar::Util qw(looks_like_number);
use Term::ANSIColor qw(:constants);


my $phonebook = "./phonebook.csv";


if( defined($ARGV[1]) && !defined($ARGV[2])){
	print "Jeżeli chcesz działac na pliku innym niz domyslny, wpisz jako trzeci arg jego nazwe. \n";
}

if( defined($ARGV[1]) && defined($ARGV[2])) {
	if ($ARGV[1] eq "-f") {
		$phonebook = "./$ARGV[2].csv"
	}
}


sub Help{
	print " \n";
	print YELLOW "Skrypt ten sluzy do tworzenia i obslugi kontaktów w ksiazce telefonicznej.\n";
	print "Jest ona zapisywana w pliku phonebook.csv i zawiera nastepujace informacje: \n";
	print "id kontaktu, nazwa, numer telefonu, kategoria, opis. \n", RESET;
	print MAGENTA"Autor: Dominika Jadach\n" , RESET;
	print YELLOW "Mozliwe dzialania: \n";
	print "--add        umozliwia dodanie kontaktu, \n";
	print "--remove     umozliwia usuniecie konkretnego kontaktu z ksiazki, \n";
	print "--removeall  usuwa wszystkie kontakty z listy, \n";
	print "--list       wypisuje liste wszystkich kontaktow, \n";
	print "--edit       umozliwia edycje danego kontaktu. \n";
	print "Aby zaczac uruchom skrypt z odpowiednim argumentem.\n";
	print "Domyslnie tworzony plik z ksiazka telefoniczna to phonebook.csv. \n";
	print "Jezeli chcesz dodac/ operowac na kontaktach z innego pliku dopisz do odpowiedniego \n";
	print " z powyzszych argumentow '-f nazwa' wedlug ponizszego przykladu dla pliku kontakty_z_pracy.csv: \n";
	print "./DominikaJadach-phonebook.pl --add -f kontakty_z_pracy \n", RESET;
}


foreach my $arg(@ARGV){
	if($arg eq "--help"){
		Help();
		exit 0;
	}
}


sub ifExists{
	my $id = shift;
	if(-e $phonebook){
		if(open(my $FH, '<', $phonebook)){
			while(my $row = <$FH>){
				chomp $row;
				my @phonebook = split(',', $row);
				if($phonebook[0] eq $id){
					return 1;
				}
			}
			close($FH);
			return 0;
		}
		else{
			print "Problem z otwarciem pliku. \n";
		}
	}
	else{
		print "Szukany plik nie istnieje. \n";
	}
}


sub endingId{
	if(open(my $FH, '<', $phonebook)){
		my $endingLine = "0";
		$endingLine = $_ while <$FH>;
		my @last = split(',', $endingLine);
		close($FH);
		return $last[0];
	}
	else{
		return "0"
	}
}


sub add{
	print BLUE "Biezacy plik: $phonebook \n", RESET;

	print YELLOW "Dodajesz kontakt. Wpisz nazwę: \n", RESET;
	my $name = <STDIN>;

	print YELLOW "Wpisz numer: \n", RESET;
	my $phoneNumber = <STDIN>;
	if (!looks_like_number($phoneNumber)){
		print RED "Ej... przeciez wiesz, ze to nie numer, podaj poprawny.\n", RESET;
		$phoneNumber = <STDIN>;
		if (looks_like_number($phoneNumber)) {
			print YELLOW "OK, udalo sie za drugim razem. \n", RESET;
		}else {
			print RED "Nie wspolpracujesz, podany numer nie jest poprawny. Nieudane dodawanie kontaktu.\n", RESET;
			exit;
		}
	}

	print YELLOW "Wybierz, do jakiej kategorii chcesz go przyporzadkowac: \n";
	print "1) sluzbowy \n";
	print "2) prywatny \n";
	print "3) inny \n", RESET;
	my $category = <STDIN>;
	my $group;
	if(!looks_like_number($category)){
		print RED "Brak kategorii. Zawsze mozesz ja dodac edytujac kontakt. \n", RESET;
		$group="";
	}else{
		if($category == 1){ $group="sluzbowy"}
		elsif($category == 2){ $group="prywatny"}
		elsif($category == 3){ $group="inny"}
		else{
			print RED "Brak kategorii. Zawsze mozesz ja dodac edytujac kontakt. \n", RESET;
			$group="";
		}
	}

	print YELLOW "Dodaj opis: \n", RESET;
	my $description = <STDIN>;

	my $id = endingId();
	chomp $id;
	chomp $name;
	chomp $phoneNumber;
	chomp $group;
	chomp $description;

	my $newid = $id + 1;

	if(open(my $FH, '>>', $phonebook)){
		print $FH "$newid,";
		print $FH "$name,";
		print $FH "$phoneNumber,";
		print $FH "$group,";
		print $FH "$description\n";
		print YELLOW "Dodano kontakt! Liste kontaktow mozesz sprawdzic uruchamiajac skrypt z argumentem --list. \n", RESET;
	}
	else{
		print RED "Problem z otwarciem pliku. \n", RESET;
	}
}


sub remove{
	print BLUE "Biezacy plik: $phonebook \n", RESET;
	if(! -z $phonebook){
		print YELLOW "Ktory kontakt chcesz usunac? Podaj jego id: \n", RESET;
		my $id = <STDIN>;
		chomp $id;

		if(ifExists($id)){
			open (FILE, "<$phonebook");
			my @PHONEBOOK = <FILE>;
			close(FILE);
			open(FILE, ">$phonebook");
			foreach my $CONTACT (@PHONEBOOK) {
				print FILE $CONTACT unless($CONTACT =~ m/^$id,/);
			}
			close(FILE);
			print RED "Kontakt zostal usuniety. \n", RESET;
		}
		else{
			print RED "Kontakt o podanym id nie istnieje, mozesz sprawdzic liste kontaktow uzywajac arg --list  \n", RESET;
		}
	}
	else{
		print RED "Brak kontaktow w ksiazce. Nie ma co usuwac. \n", RESET;
		exit 0;
	}
}


sub removeall{
	print BLUE "Biezacy plik: $phonebook \n", RESET;
	if(! -z $phonebook){
		open(my $FH, '>', $phonebook);
		print RED "Usunieto wszystkie kontakty. \n", RESET;
	}
	else{
		print RED "Brak kontaktow. \n", RESET;
	}
}


sub list{
	print BLUE "Biezacy plik: $phonebook \n", RESET;
	if(-z $phonebook){
		print RED "Brak kontaktow, lista jest pusta. \n", RESET;
	}else{
		if(open(my $FH, '<', $phonebook)){
			print YELLOW "Dane kontaktów wypisane w ponizszej kolejności: \n";
			print "ID, nazwa, numer, kategoria, opis \n";
			print "--------------------------------- \n", RESET;
			while(my $row = <$FH>){
				chomp $row;
				my @contact = split(',', $row);
				print GREEN" $contact[0], $contact[1], $contact[2], $contact[3], $contact[4] \n", RESET;
			}
			close($FH);
		}
	}
}


sub isOne{
	my $line;
	my $lines = 1;
	open(my $fh, '<', $phonebook) or die "Nie da się otworzyc pliku: $phonebook: $!";
	$line = <$fh>;
	$lines++ while <$fh>;
	close $fh;

	if ($lines eq 1){
		return 1;
	}
	else {
		return 0;
	}
}


sub edit{
	print BLUE "Biezacy plik: $phonebook \n", RESET;

	my $id = 0;
	if(-z $phonebook){
		print RED "Brak kontaktow, lista jest pusta. \n", RESET;
	}
	else{
		if(isOne){
			print YELLOW "Nie masz wielkiego wyboru, w Twojej ksiazce znajduje sie jeden kontakt. \n", RESET;
			if(open(my $FH, '<', $phonebook)){
				while(my $row = <$FH>){
					chomp $row;
					my @contact = split(',', $row);
					$id = $contact[0];
				}
				close($FH);
			}
		}
		else {
			print YELLOW "Ktory kontakt chcesz edytowac? Podaj jego id: \n", RESET;
			$id = <STDIN>;
			chomp $id;
		}

		if(ifExists($id)){
			print YELLOW "Co chcesz edytowac? Wybierz odpowiednia akcje: \n";
			print "1) Zmiana nazwy \n";
			print "2) Zmiana numeru \n";
			print "3) Zmiana kategorii \n";
			print "4) Zmiana opisu \n", RESET;
			my $selection = <STDIN>;

			if($selection == 1){
				print YELLOW "Nowa nazwa: \n", RESET;
				my $newName = <STDIN>;
				chomp $newName;
				open(FILE, "<$phonebook");
				my @PHONEBOOK = <FILE>;
				close(FILE);
				open(FILE, ">$phonebook");
				foreach my $PHONOBOOK (@PHONEBOOK) {
					if ($PHONOBOOK =~ m/^$id,/) {
						my @splited = split(',', $PHONOBOOK);
						$splited[1] = $newName;
						my $newPhonebook = join(',', @splited);
						print FILE $newPhonebook;
					} else {
						print FILE $PHONOBOOK;
					}
				}
				close(FILE);
			}

			elsif($selection == 2){
				print YELLOW "Nowy numer: \n", RESET;
				my $newNumber = <STDIN>;
				if(!looks_like_number($newNumber)){
					print RED "Ej... przeciez wiesz, ze to nie numer, podaj poprawny, zeby edytowac.\n", RESET;
					$newNumber = <STDIN>;
					if(looks_like_number($newNumber)){
						print YELLOW "OK, udalo sie za drugim razem.\n", RESET;
					}else{
						print RED "Nie wspolpracujesz, podany numer nie jest poprawny. Nieudane edytowanie kontaktu.\n", RESET;
						exit(0);
					}
				}
				chomp $newNumber;
				open(FILE, "<$phonebook");
				my @PHONEBOOK = <FILE>;
				close(FILE);
				open(FILE, ">$phonebook");
				foreach my $PHONOBOOK (@PHONEBOOK) {
					if ($PHONOBOOK =~ m/^$id,/) {
						my @splited = split(',', $PHONOBOOK);
						$splited[2] = $newNumber;
						my $newPhonebook = join(',', @splited);
						print FILE $newPhonebook;
					} else {
						print FILE $PHONOBOOK;
					}
				}
				close(FILE);
			}

			elsif($selection == 3){
				my $group;
				print YELLOW "Wybierz nowa kategorie \n";
				print "1) sluzbowy \n";
				print "2) prywatny \n";
				print "3) inny \n", RESET;
				my $category = <STDIN>;
				if($category == 1){
					$group = "sluzbowy";
				}
				elsif($category == 2){
					$group = "prywatny";
				}
				elsif($category == 3){
					$group = "inny";
				}
				else{
					print RED "Brak kategorii. Zawsze mozesz ja dodac edytujac kontakt.\n", RESET;
					$group = "";
				}
				chomp $group;
				open(FILE, "<$phonebook");
				my @PHONEBOOK = <FILE>;
				close(FILE);
				open(FILE, ">$phonebook");
				foreach my $PHONOBOOK (@PHONEBOOK) {
					if ($PHONOBOOK =~ m/^$id,/) {
						my @splited = split(',', $PHONOBOOK);
						$splited[3] = $group;
						my $newPhonebook = join(',', @splited);
						print FILE $newPhonebook;
					} else {
						print FILE $PHONOBOOK;
					}
				}
				close(FILE);
			}

			elsif($selection == 4){
				print YELLOW "Nowy opis: \n", RESET;
				my $newDescription = <STDIN>;
				chomp $newDescription;
				open(FILE, "<$phonebook");
				my @PHONEBOOK = <FILE>;
				close(FILE);
				open(FILE, ">$phonebook");
				foreach my $PHONOBOOK (@PHONEBOOK) {
					if ($PHONOBOOK =~ m/^$id,/) {
						my @splited = split(',', $PHONOBOOK);
						$splited[4] = $newDescription;
						my $newPhonebook = join(',', @splited);
						print FILE $newPhonebook;
					} else {
						print FILE $PHONOBOOK;
					}
				}
				close(FILE);
			}
			else{
				print RED "Cos Ci sie pomylilo, mozliwe wybory to 1, 2, 3 lub 4. \n", RESET;
			}
		}
		else{
			print RED "Kontakt o podanym id nie istnieje, mozesz sprawdzic liste kontaktow uzywajac arg --list \n", RESET;
		}
	}
}


sub optionsToChoose{
	print RED "Nieznany argument! \n";
	print "Oto te, które znam: \n";
	print "--add \n";
	print "--remove \n";
	print "--removeall \n";
	print "--list \n";
	print "--edit \n", RESET;
}


if(! defined $ARGV[0]){
	optionsToChoose();
}
elsif($ARGV[0] eq "--add"){
	add();
}
elsif($ARGV[0] eq "--remove"){
	remove()
}
elsif($ARGV[0] eq "--removeall"){
	removeall();
}
elsif($ARGV[0] eq "--list"){
	list();
}
elsif($ARGV[0] eq "--edit"){
	edit();
}
else{
	optionsToChoose();
	exit 0;
}

File: phonebook.py
#!/usr/bin/python3
# Jadach Dominika - projekt zaliczeniowy - grupa 1


import sys
import os.path
import csv

phonebook = "./phonebook.csv"


class colors:
    MAGENTA = '\033[95m'
    BLUE = '\033[94m'
    GREEN = '\033[92m'
    YELLOW = '\033[93m'
    RED = '\033[91m'
    ENDC = '\033[0m'


class Phonebook:

    def Help(self):
        print(colors.YELLOW + "Skrypt ten sluzy do tworzenia i obslugi kontaktow w ksiazce telefonicznej.")
        print("Jest ona zapisywana w pliku phonebook.csv i zawiera nastepujace informacje: ")
        print("id kontaktu, nazwa, numer telefonu, kategoria, opis. " + colors.ENDC)
        print(colors.MAGENTA + "Autor: Dominika Jadach - grupa 1 " + colors.ENDC)
        print(colors.YELLOW + "Mozliwe dzialania: ")
        print("--add        umozliwia dodanie kontaktu, ")
        print("--remove     umozliwia usuniecie konkretnego kontaktu z ksiazki, ")
        print("--removeall  usuwa wszystkie kontakty z listy, ")
        print("--list       wypisuje liste wszystkich kontaktow, ")
        print("--edit       umozliwia edycje danego kontaktu. ")
        print("Aby zaczac uruchom skrypt z odpowiednim argumentem.")
        print("Domyslnie tworzony plik z ksiazka telefoniczna to phonebook.csv. ")
        print("Jezeli chcesz dodac/ operowac na kontaktach z innego pliku dopisz do odpowiedniego ")
        print(" z powyzszych argumentow '-f nazwa' wedlug ponizszego przykladu dla pliku kontakty_z_pracy.csv: ")
        print("./DominikaJadach-phonebook.py --add -f kontakty_z_pracy " + colors.ENDC)

    def ifExists(self, id):
        if os.path.isfile(phonebook):
            with open(phonebook) as csv_file:
                csv_reader = csv.reader(csv_file, delimiter=',')
                for row in csv_reader:
                    if row[0] == id:
                        return 1
        else:
            print("Szukany plik nie istnieje.")
            return 0

    def endingId(self):
        ending_id = 0
        with open(phonebook) as csv_file:
            csv_reader = csv.reader(csv_file, delimiter=',')
            for row in csv_reader:
                ending_id = row[0]
        ret = int(ending_id) + 1
        return ret

    def add(self):
        print(colors.BLUE + "Biezacy plik: " + phonebook + colors.ENDC)
        name = input(colors.YELLOW + "Dodajesz kontakt. Wpisz nazwe: " + colors.ENDC)
        number = input(colors.YELLOW + "Wpisz numer: " + colors.ENDC)
        if not number.isnumeric():
            print(colors.RED + "Ej... przeciez wiesz, ze to nie numer, podaj poprawny." + colors.ENDC)
            number = input()
            if not number.isnumeric():
                print(colors.RED + "Nie wspolpracujesz, podany numer nie jest poprawny. Nieudane dodawanie kontaktu." + colors.ENDC)
                return
            else:
                print(colors.YELLOW + "OK, udalo sie za drugim razem." + colors.ENDC)
        print(colors.YELLOW + "Wybierz, do jakiej kategorii chcesz go przyporzadkowac: ")
        print("1) sluzbowy")
        print("2) prywatny")
        print("3) inny" + colors.ENDC)
        category = input()
        if category == "1":
            group = "sluzbowy"
        elif category == "2":
            group = "prywatny"
        elif category == "3":
            group = "inny"
        else:
            print(colors.RED + "Brak kategorii. Zawsze mozesz ja dodac edytujac kontakt." + colors.ENDC)
            group = ""
        description = input(colors.YELLOW + "Dodaj opis: " + colors.ENDC)
        id = self.endingId()
        with open(phonebook, 'a+', newline='') as csv_file:
            csv_writer = csv.writer(csv_file)
            csv_writer.writerow([id, name, number, group, description])
            print(colors.YELLOW + "Dodano kontakt! Liste kontaktow mozesz sprawdzic uruchamiajac skrypt z argumentem --list." + colors.ENDC)

    def remove(self):
        print(colors.BLUE + "Biezacy plik: " + phonebook + colors.ENDC)
        if os.stat(phonebook).st_size == 0:
            print(colors.RED + "Brak kontaktow, lista jest pusta. " + colors.ENDC)
            return
        id = input(colors.YELLOW + "Ktory kontakt chcesz usunac? Podaj jego id: " + colors.ENDC)
        if self.ifExists(id):
            lines = list()
            with open(phonebook) as csv_file:
                csv_reader = csv.reader(csv_file, delimiter=',')
                for row in csv_reader:
                    lines.append(row)
                    for field in row:
                        if field == id:
                            lines.remove(row)
                            print(colors.RED + "Kontakt zostal usuniety. " + colors.ENDC)
                        else:
                            continue
            with open(phonebook, 'w') as csv_file:
                csv_writer = csv.writer(csv_file, delimiter=',')
                csv_writer.writerows(lines)
        else:
            print(colors.RED + "Kontakt o podanym id nie istnieje, mozesz sprawdzic liste kontaktow uzywajac arg --list" + colors.ENDC)

    def removeall(self):
        print(colors.BLUE + "Biezacy plik: " + phonebook + colors.ENDC)
        if os.stat(phonebook).st_size == 0:
            print(colors.RED + "Brak kontaktow, lista jest pusta. " + colors.ENDC)
            return
        f = open(phonebook, "w+")
        f.close()
        print(colors.RED + "Usunieto wszystkie kontakty." + colors.ENDC)

    def list(self):
        print(colors.BLUE + "Biezacy plik: " + phonebook + colors.ENDC)
        if os.stat(phonebook).st_size == 0:
            print(colors.RED + "Brak kontaktow, lista jest pusta. " + colors.ENDC)
            return
        with open(phonebook) as csv_file:
            csv_reader = csv.reader(csv_file, delimiter=',')
            line_count = 0
            print(colors.YELLOW + "Dane kontaktów wypisane w ponizszej kolejności: ")
            print("ID, nazwa, numer, kategoria, opis")
            print("---------------------------------" + colors.ENDC)
            for row in csv_reader:
                print(colors.GREEN + row[0], ",", row[1], ",", row[2], ",", row[3], ",", row[4] + colors.ENDC)
                line_count += 1

    def isOne(self):
        with open(phonebook) as csv_file:
            csv_reader = csv.reader(csv_file, delimiter=',')
            line_count = 0
            for _ in csv_reader:
                line_count += 1
        if line_count == 1:
            return 1
        else:
            return 0

    def edit(self):
        print(colors.BLUE + "Biezacy plik: " + phonebook + colors.ENDC)
        if os.stat(phonebook).st_size == 0:
            print(colors.RED + "Brak kontaktow, lista jest pusta. " + colors.ENDC)
            return
        if(self.isOne()):
            print(colors.YELLOW + "Nie masz wielkiego wyboru, w Twojej ksiazce znajduje sie jeden kontakt." + colors.ENDC)
            with open(phonebook) as csv_file:
                csv_reader = csv.reader(csv_file, delimiter=',')
                for row in csv_reader:
                    id = row[0]
        else:
            print(colors.YELLOW + "Ktory kontakt chcesz edytowac? Podaj jego id: " + colors.ENDC)
            id = input()
        lines = list()
        if self.ifExists(id):
            print(colors.YELLOW + "Co chcesz edytowac? Wybierz odpowiednia akcje: ")
            print("1) Zmiana nazwy")
            print("2) Zmiana numeru")
            print("3) Zmiana kategorii")
            print("4) Zmiana opisu" + colors.ENDC)
            selection = input()
        else:
            print(colors.RED + "Kontakt o podanym id nie istnieje, mozesz sprawdzic liste kontaktow uzywajac arg --list" + colors.ENDC)
            return
        if selection == "1":
            newName = input(colors.YELLOW + "Nowa nazwa: " + colors.ENDC)
            with open(phonebook) as csv_file:
                csv_reader = csv.reader(csv_file, delimiter=',')
                for row in csv_reader:
                    lines.append(row)
                    for field in row:
                        if field == id:
                            row[1] = newName
            with open(phonebook, 'w') as csv_file:
                csv_writer = csv.writer(csv_file, delimiter=',')
                csv_writer.writerows(lines)
        elif selection == "2":
            newNumber = input(colors.YELLOW + "Nowy numer: " + colors.ENDC)
            if not newNumber.isnumeric():
                print(colors.RED + "Ej... przeciez wiesz, ze to nie numer, podaj poprawny." + colors.ENDC)
                newNumber = input()
                if not newNumber.isnumeric():
                    print(colors.RED + "Nie wspolpracujesz, podany numer nie jest poprawny. Nieudane dodawanie kontaktu." + colors.ENDC)
                    return
                else:
                    print(colors.YELLOW + "OK, udalo sie za drugim razem." + colors.ENDC)
            with open(phonebook) as csv_file:
                csv_reader = csv.reader(csv_file, delimiter=',')
                for row in csv_reader:
                    lines.append(row)
                    for field in row:
                        if field == id:
                            row[2] = newNumber
            with open(phonebook, 'w') as csv_file:
                csv_writer = csv.writer(csv_file, delimiter=',')
                csv_writer.writerows(lines)
        elif selection == "3":
            print(colors.YELLOW + "Wybierz nowa kategorie: ")
            print("1) sluzbowy")
            print("2) prywatny")
            print("3) inny" + colors.ENDC)
            category = input()
            if category == "1":
                group = "sluzbowy"
            elif category == "2":
                group = "prywatny"
            elif category == "3":
                group = "inny"
            else:
                group = ""
                print(colors.RED + "Brak kategorii. Zawsze mozesz ja dodac edytujac kontakt." + colors.ENDC)
            with open(phonebook) as csv_file:
                csv_reader = csv.reader(csv_file, delimiter=',')
                for row in csv_reader:
                    lines.append(row)
                    for field in row:
                        if field == id:
                            row[3] = group
            with open(phonebook, 'w') as csv_file:
                csv_writer = csv.writer(csv_file, delimiter=',')
                csv_writer.writerows(lines)
        elif selection == "4":
            newDescription = input(colors.YELLOW + "Nowy opis: " + colors.ENDC)
            with open(phonebook) as csv_file:
                csv_reader = csv.reader(csv_file, delimiter=',')
                for row in csv_reader:
                    lines.append(row)
                    for field in row:
                        if field == id:
                            row[4] = newDescription
            with open(phonebook, 'w') as csv_file:
                csv_writer = csv.writer(csv_file, delimiter=',')
                csv_writer.writerows(lines)
        else:
            print(colors.RED + "Cos Ci sie pomylilo, mozliwe wybory to 1, 2, 3 lub 4." + colors.ENDC)
            return

    def optionsToChoose(self):
        print(colors.RED + "Nieznany argument! ")
        print("Oto te, ktore znam:")
        print("--add ")
        print("--remove ")
        print("--removeall ")
        print("--list ")
        print("--edit " + colors.ENDC)


book = Phonebook()

if len(sys.argv) == 3:
    print("Jezeli chcesz dzialac na pliku innym niz domyslny, wpisz jako trzeci arg jego nazwe.")

elif len(sys.argv) == 4:
    if sys.argv[2] == "-f":
        phonebook = "./" + sys.argv[3] + ".csv"

if len(sys.argv) > 1 and sys.argv[1] == "--help":
    Phonebook.Help(book)
elif len(sys.argv) > 1 and sys.argv[1] == "--add":
    Phonebook.add(book)
elif len(sys.argv) > 1 and sys.argv[1] == "--remove":
    Phonebook.remove(book)
elif len(sys.argv) > 1 and sys.argv[1] == "--removeall":
    Phonebook.removeall(book)
elif len(sys.argv) > 1 and sys.argv[1] == "--list":
    Phonebook.list(book)
elif len(sys.argv) > 1 and sys.argv[1] == "--edit":
    Phonebook.edit(book)
else:
    Phonebook.optionsToChoose(book)
    exit(0)
